### ISSUE TITLE
How many training examples are you use?

### ISSUE THREAD

**Issue by liujingcs** [foreverYoungGitHub/MTCNN#24]
How many training examples are you use? I extract 6 non-face patches, 2 part-face patches, and 2 positive patches every image, so I get totally 1 billion training data. Is this too many?

**Comment by foreverYoungGitHub** [comment]
Which dataset did you use?

While, having plenty of faces is a good thing. But you need enough time to train it.

In caffe, each iteration updates the weights based on the input batches. When the number of the batches are 100.

1 billion training data means you need to train 10^9 / 100 = 10 ^ 7 times iteration to let all the data training for one time.

Waiting for long time .....

In fact, in computer vision, the most classification task always trains for 20K ~ 40K to achieved the baseline. And fine-tuning the model based on bootstrap. You can try that.

By the way, I used near 2 M samples for the first time training.

**Comment by liujingcs** [comment]
Thanks for your response. I use CelebA data set. I have some question. 
1. How to set learning rate? I found that large learning rate will result to NaN. 
2. How to train in fine-tuning stage? Is the training data the same proportion as previous training?
